pandas/io/clipboard/__init__.py

```diff
@@ -10,116 +11,693 @@
   pyperclip.copy('The text to be copied to the clipboard.')
   spam = pyperclip.paste()
 
-  if not pyperclip.copy:
+  if not pyperclip.is_available():
     print("Copy functionality unavailable!")
 
 On Windows, no additional modules are needed.
-On Mac, the module uses pbcopy and pbpaste, which should come with the os.
+On Mac, the pyobjc module is used, falling back to the pbcopy and pbpaste cli
+    commands. (These commands should come with OS X.).
 On Linux, install xclip or xsel via package manager. For example, in Debian:
-sudo apt-get install xclip
+    sudo apt-get install xclip
+    sudo apt-get install xsel
 
-Otherwise on Linux, you will need the qtpy or PyQt modules installed.
-qtpy also requires a python-qt-bindings module: PyQt4, PyQt5, PySide, PySide2
+Otherwise on Linux, you will need the gtk or PyQt5/PyQt4 modules installed.
+
+gtk and PyQt4 modules are not available for Python 3,
+and this module does not work with PyGObject yet.
+
+Note: There seems to be a way to get gtk on Python 3, according to:
+    https://askubuntu.com/questions/697397/python3-is-not-supporting-gtk-module
+
+Cygwin is currently not supported.
+
+Security Note: This module runs programs with these names:
+    - which
+    - where
+    - pbcopy
+    - pbpaste
+    - xclip
+    - xsel
+    - klipper
+    - qdbus
+A malicious user could rename or add programs with these names, tricking
+Pyperclip into running them with whatever permissions the Python process has.
 
-This module does not work with PyGObject yet.
 """
-__version__ = "1.5.27"
+__version__ = "1.7.0"
 
+import contextlib
+import ctypes
+from ctypes import c_size_t, c_wchar, c_wchar_p, get_errno, sizeof
 import os
 import platform
 import subprocess
+import time
+import warnings
 
-from .clipboards import (
-    init_klipper_clipboard,
-    init_no_clipboard,
-    init_osx_clipboard,
-    init_qt_clipboard,
-    init_xclip_clipboard,
-    init_xsel_clipboard,
-)
-from .windows import init_windows_clipboard
-
-# `import qtpy` sys.exit()s if DISPLAY is not in the environment.
+# `import PyQt4` sys.exit()s if DISPLAY is not in the environment.
```

Comment:
is this needed?

Reply:
What do you mean by *this*?

I only altered the original pyperclip 1.7.0 code as much as it was necessary

Reply:
i mistook this comment for commented-out code, my mistake.